peekaboo/toolbox/cuckoo.py

```diff
@@ -40,47 +43,227 @@
-        logger.debug("cuckoo submit STDERR: %s" % err)
-        # process output to get job ID
-        patterns = list()
-        # Example: Success: File "/var/lib/peekaboo/.bashrc" added as task with ID #4
-        patterns.append(".*Success.*: File .* added as task with ID #([0-9]*).*")
-        patterns.append(".*added as task with ID ([0-9]*).*")
-        matcher = MultiRegexMatcher(patterns)
-        response = out.replace("\n", "")
-        m = matcher.match(response)
-        logger.debug('Pattern %d matched.' % matcher.matched_pattern)
+    def __init__(self, interpreter, cuckoo_exec):
+        self.interpreter = interpreter
+        self.cuckoo_exec = cuckoo_exec
+    
+    def submit(self, sample):
+        """
+            Submit a file or directory to Cuckoo for behavioural analysis.
+            
+            :param sample: Path to a file or a directory.
+            :return: The job ID used by Cuckoo to identify this analysis task.
+            """
+        config = get_config()
+        try:
+            proc = config.cuckoo_submit
+            proc.append(sample)
+            p = subprocess.Popen(proc,
+                                 stdout=subprocess.PIPE,
+                                 stderr=subprocess.PIPE)
+            p.wait()
+        except Exception as e:
+            raise CuckooAnalysisFailedException(e)
+        
+        if not p.returncode == 0:
+            # TODO: tell opponent on socket that file has not been checked.
+            raise CuckooAnalysisFailedException('cuckoo submit returned a non-zero return code.')
+        else:
+            out, err = p.communicate()
+            logger.debug("cuckoo submit STDOUT: %s" % out)
+            logger.debug("cuckoo submit STDERR: %s" % err)
+            # process output to get job ID
+            patterns = list()
+            # Example: Success: File "/var/lib/peekaboo/.bashrc" added as task with ID #4
+            patterns.append(".*Success.*: File .* added as task with ID #([0-9]*).*")
+            patterns.append(".*added as task with ID ([0-9]*).*")
+            matcher = MultiRegexMatcher(patterns)
+            response = out.replace("\n", "")
+            m = matcher.match(response)
+            logger.debug('Pattern %d matched.' % matcher.matched_pattern)
+            
+            if m:
+                job_id = int(m.group(1))
+                return job_id
+            raise CuckooAnalysisFailedException(
+                                                'Unable to extract job ID from given string %s' % response
+                                                )
+
+    def do(self):
+        # reaktor and shit
+        # Run Cuckoo sandbox, parse log output, and report back of Peekaboo.
+        srv = CuckooServer()
+        reactor.spawnProcess(srv, self.interpreter, [self.interpreter, '-u',
+                                                     self.cuckoo_exec])
+        reactor.run()
+
+
+class CuckooDummy(Cuckoo):
```

Comment:
``CuckooDummy`` is for development purposes only. So, it should be a part of the unit tests or a separate script.